crates/red_knot_python_semantic/mdtest.py

```diff
@@ -0,0 +1,200 @@
+"""A runner for Markdown-based tests for Red Knot"""
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#     "rich",
+#     "watchfiles",
+# ]
+# ///
+
+import json
+import os
+import subprocess
+from pathlib import Path
+from typing import Literal, Never
+
+from rich.console import Console
+from watchfiles import Change, watch
+
+CRATE_NAME = "red_knot_python_semantic"
+CRATE_ROOT = Path(__file__).resolve().parent
+MDTEST_DIR = CRATE_ROOT / "resources" / "mdtest"
```

Comment:
minor: you could annotate these with `Final`:

```suggestion
from typing import Final, Literal, Never

from rich.console import Console
from watchfiles import Change, watch

CRATE_NAME: Final = "red_knot_python_semantic"
CRATE_ROOT: Final = Path(__file__).resolve().parent
MDTEST_DIR: Final = CRATE_ROOT / "resources" / "mdtest"
```